### PR TITLE
fix: env var | cypress | dependencies | ui styles

### PR DESCRIPTION
**Pull request by 0xtiti**
closes FRO-8

PR #7 fixes

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
-import { defineConfig } from "cypress";
+import { defineConfig } from 'cypress';
 
 export default defineConfig({
   e2e: {
-    setupNodeEvents(on, config) {
-      // implement node event listeners here
-    },
+    baseUrl: 'http://localhost:3000',
```
Comment (turtlemoji) [inline cypress.config.ts:5]: currently the build is broken, but also the tests, it is not running as expected the test action.

File: src/config/env.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 
 export const getEnv = (): Env => {
-  const { NEXT_PUBLIC_RPC_URL, NEXT_PUBLIC_PROJECT_ID, NEXT_PUBLIC_ALCHEMY_KEY } = process.env;
+  const NEXT_PUBLIC_RPC_URL = process.env.NEXT_PUBLIC_RPC_URL;
+  const NEXT_PUBLIC_PROJECT_ID = process.env.NEXT_PUBLIC_PROJECT_ID;
+  const NEXT_PUBLIC_ALCHEMY_KEY = process.env.NEXT_PUBLIC_ALCHEMY_KEY;
+
+  if (!NEXT_PUBLIC_RPC_URL || !NEXT_PUBLIC_PROJECT_ID || !NEXT_PUBLIC_ALCHEMY_KEY) {
+    throw new Error('Environment configuration is missing critical values');
+  }
```
Comment (turtlemoji) [inline src/config/env.ts]: We can remove it for now so it doesn't stop the build without env params. And you can leave the constants as they are right now (without the `|| ''`)

**Review by turtlemoji — CHANGES_REQUESTED** [review]
recheck build

File: cypress.config.ts (hunk 1 above)
Comment (0xtiti) [reply cypress.config.ts:5]: [3867448](https://github.com/defi-wonderland/web3-react-boilerplate/pull/8/commits/3867448d15ac4b9e4b11d341e62272b4db9855c9)

File: src/config/env.ts (hunk 2 above)
Comment (0xtiti) [reply src/config/env.ts]: https://github.com/defi-wonderland/web3-react-boilerplate/pull/8/commits/7493046924de6f9df71697df242ea2df004c01b3

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -58,32 +50,29 @@ jobs:
...
       - name: Run Linter
-        run: yarn lint
+        run: pnpm lint
 
-      - name: Run Tests
-        run: yarn test
+      - name: Cypress run
+        uses: cypress-io/github-action@v5
+        with:
+          start: pnpm start, pnpm test
```
Comment (turtlemoji) [inline .github/workflows/build-test.yml:78]: please check @0xArdy too

Comment (turtlemoji) [reply .github/workflows/build-test.yml:78]: (idk much about cypress best practices into actions)

Comment (0xArdy) [reply .github/workflows/build-test.yml:78]: yeap, that's what Nextjs recommends to use with Cypress

> Cypress will simulate a user navigating your application, this requires your Next.js server to be running. We recommend running your tests against your production code to more closely resemble how your application will behave.

https://nextjs.org/docs/pages/building-your-application/testing/cypress#running-e2e-tests

**Review by 0xArdy — APPROVED** [review]
lgtm!

File: src/utils/Variables.ts [hunk 4]
```diff
@@ -20,3 +20,14 @@ export const fontSize = {
...
+export const zIndex = {
+  HEADER: 100,
+  MODAL: 200,
+  BACKDROP: -1,
+  TOAST: 500,
+};
+
+export const HEADER_HEIGHT = 8; // Header height in rem units
+export const FOOTER_HEIGHT = 8; // Footer height in rem units
+export const MAIN_HEIGHT = HEADER_HEIGHT + FOOTER_HEIGHT;
```
Comment (agusduha) [inline src/utils/Variables.ts:33]: This is more like the opposite of the main height right? Like a discounter

Comment (0xShaito) [reply src/utils/Variables.ts:33]: Same Q. Im curious about the naming for it

**Review by 0xShaito — APPROVED** [review]

**Review by turtlemoji — APPROVED** [review]

File: src/utils/Variables.ts (hunk 4 above)
Comment (0xArdy) [reply src/utils/Variables.ts:33]: Yep, it isnt very clear, we can fix it in a next pr

**Pull request merged**